agglomcluster/agglomod.py

```diff
@@ -155,15 +161,24 @@ def add_pair_to_cost_heap(self, id1, id2):
 
     # The "Change in Q" as described by section II of the Newman paper
     def quality_difference(self, cluster_id1, cluster_id2):
-        ai = float(self.super_graph.node[cluster_id1])
-        aj = float(self.super_graph.node[cluster_id2])
+        if self.unique_string:
+            if self.super_graph.node[cluster_id1][self.unique_string] and self.super_graph.node[cluster_id2][self.unique_string]:
+                return 0
+        ai = float(self.super_graph.node[cluster_id1]['degree'])
+        aj = float(self.super_graph.node[cluster_id2]['degree'])
         eij = float(self.super_graph[cluster_id1][cluster_id2])
         return 2.0*(eij - ai*aj)
 
     def combine_clusters(self, cluster_id1, cluster_id2):
         combine_id = self.den_num
         self.den_num += 1
-        
+
+        flag = False
+        # if either have a flag then make sure the combined cluster also gets the flag
+        if self.unique_string:
+            if self.super_graph.node[cluster_id1][self.unique_string] or self.super_graph.node[cluster_id2][self.unique_string]:
+                flag = True

```

Comment:
Can do this more concisely:

flag = self.unique_string and (self.super_graph.node[cluster_id1][self.unique_string] or self.super_graph.node[cluster_id2][self.unique_string])